include/coap2/net.h

```diff
@@ -844,4 +724,265 @@ unsigned int coap_calc_timeout(coap_session_t *session, unsigned char r);
 int
 coap_join_mcast_group(coap_context_t *ctx, const char *groupname);
 
+/**
+ * @defgroup app_io Application I/O Handling
+ * API functions for Application Input / Output
+ * @{
+ */
+
+#define COAP_RUN_BLOCK    0
+#define COAP_RUN_NONBLOCK 1
+
+/**
+ * The main I/O processing function.  All pending network I/O is completed,
+ * and then optionally waits for the next input packet.
+ *
+ * This internally calls coap_io_prepare_io(), then select() for the appropriate
+ * sockets, updates COAP_SOCKET_CAN_xxx where appropriate and then calls
+ * coap_io_do_io() before returning with the time spent in the function.
+ *
+ * Alternatively, if libcoap is compiled with epoll support, this internally
+ * calls coap_io_prepare_epoll(), then epoll_wait() for waiting for any file
+ * descriptors that have (internally) been set up with epoll_ctl() and
+ * finally coap_io_do_epoll() before returning with the time spent in the
+ * function.
+ *
+ * @param ctx The CoAP context
+ * @param timeout_ms Minimum number of milliseconds to wait for new packets
+ *                   before returning. If COAP_RUN_BLOCK, the call will block
+ *                   until at least one new packet is received. If
+ *                   COAP_RUN_NONBLOCK, the function will return immediately
+ *                   following without waiting for any new input not already
+ *                   available.
+ *
+ * @return Number of milliseconds spent in function or @c -1 if there was
+ *         an error
+ */
+int coap_io_process(coap_context_t *ctx, unsigned int timeout_ms);
+
+/**
+ * @deprecated Use coap_io_process() instead.
+ *
+ * The main I/O processing function.  All pending network I/O is completed,
+ * and then optionally waits for the next input packet.
+ *
+ * This internally calls coap_io_prepare_io(), then select() for the appropriate
```

Comment:
I suggest documenting here that this function internally calls `coap_io_process()` to avoid inconsistent descriptions of the internal `coap_io_process()` functionality.

Reply:
Updated as suggested.